### ISSUE TITLE
docs: create guide to test pull requests

### ISSUE THREAD

**Issue by gaurivn** [anitab-org/vms#1019]
## Description
As a contributor for testing,
I need a guide to help me understand how to test pull requests on VMS,
so that I can properly test pull requests from contributors that solve issues for the same.

### Update [Required]
- [ ] Create guide on docs folder, name it test-pr-guide.md

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

**Comment by ashwinijha6** [comment]
Hi @gaurivn I would like to work on this issue

**Comment by gaurivn** [comment]
Hi @ashwinijha6, glad to know you're interested in working on this issue. This issue is not made available yet, it will be made available once it's approved and Open Source Hack starts on September 1st, 2020. Till then you can set up the development environment for and familiarise yourself with VMS. If you've any doubts feel free to ask them on Zulip.

**Comment by vaibhavarora102** [comment]
Hi @gaurivn  can anyone participate in Open Source Hack 
can you please regarding this as I am a bit new to Open Source
thanks

**Comment by ashwinijha6** [comment]
Yes sure @gaurivn I will be participating in OSH. I didn't notice that the status was on hold. Eager to work on this issue when it will be made available & when OSH starts.

**Comment by SanketDG** [comment]
@gaurivn Do you mean to say instructions for running the application?

**Comment by gaurivn** [comment]
> Hi @gaurivn can anyone participate in Open Source Hack
> can you please regarding this as I am a bit new to Open Source
> thanks

Hi @vaibhavarora102, yes anyone (students and working professionals) can participate in Open Source Hack). The aim is to help participants contribute to various open-source projects.

**Comment by gaurivn** [comment]
> @gaurivn Do you mean to say the README should provide instructions for running the application?

No, there are separate instructions for installation right.

**Comment by vaibhavarora102** [comment]
@gaurivn So can you pls assign me as I would like to work on this in Open source hack 
thanks

**Comment by SanketDG** [comment]
Feel free to provide an example of what this should look like @gaurivn

**Comment by gaurivn** [comment]
> @gaurivn So can you pls assign me as I would like to work on this in Open source hack
> thanks

@vaibhavarora102, the issues meant for open source hack will be assigned only when the program starts on September 1st, 2020 on a first come first serve basis.

**Comment by gaurivn** [comment]
> Feel free to provide an example of what this should look like @gaurivn

I've updated the issue description.

**Comment by ashwinijha6** [comment]
@gaurivn I would like to work on this issue as an OSH participant

**Comment by vaibhavarora102** [comment]
Hi @gaurivn , can you pls assign this issue to me as its already 1st of September and I guess OSH has started
Thanks

**Comment by Abjcodes** [comment]
I would like to work on this issue as a part of OSH. Can you assign me this issue? Thank You.

**Comment by gaurivn** [comment]
> @gaurivn I would like to work on this issue as an OSH participant

Sure, I've assigned you, feel free to work on it and ask doubts if any here or on Zulip :)

**Comment by gaurivn** [comment]
> Hi @gaurivn , can you pls assign this issue to me as its already 1st of September and I guess OSH has started
> Thanks

Hi @vaibhavarora102, yes.. you've already been assigned issue #1028. Once you send a PR for that and if this issue is available to work on at that time, I'll assign you.

**Comment by gaurivn** [comment]
> I would like to work on this issue as a part of OSH. Can you assign me this issue? Thank You.

Hi @Abjcodes, you've already been assigned issue #1026, once you send a PR for that and if this issue is available at that time, will assign.

**Comment by titiksha01** [comment]
@gaurivn  Hello, I would like to work on this issue as a part of OSH. Can this be assigned to me? Thank you

**Comment by vaibhavarora102** [comment]
@gaurivn I am done with the previous issue
and hope i will now be assigned the issue

**Comment by ashwinijha6** [comment]
@gaurivn I had a doubt related to this as I have started working on it should the test-pr-guide.md contain only the 10 steps as mentioned in the mock?

**Comment by gaurivn** [comment]
> @gaurivn I had a doubt related to this as I have started working on it should the test-pr-guide.md contain only the 10 steps as mentioned in the mock?

No, not just the 10 steps, I've edited the issue description to include the link of the corresponding file in mentorship-backend. Pls, add the debugging steps too, from the admin side like entering, viewing volunteer contact info, tracking volunteer hours, and assigning a volunteer to a job.

**Comment by vaibhavarora102** [comment]
hi @gaurivn  can you pls assign this issue to me as I had already completed all my previous issues assigned
thanks

**Comment by gaurivn** [comment]
> hi @gaurivn can you pls assign this issue to me as I had already completed all my previous issues assigned
> thanks

@vaibhavarora102, since this issue is already assigned, can't assign again for the time being. In case this issue becomes available in the future will assign you.

**Comment by Anmollenka** [comment]
Can I work on this issue as a part of OSH??Thank you

**Comment by ashwinijha6** [comment]
@gaurivn could u pls help me, I understand that the test pull request guide should contain 
1. Steps to test a PR
2. Template to report PR testing results
3. Steps to test multiple features (I cannot understand what I need to add in this part )
Do u mean that need to run each test using **python manage.py test** & provide the screenshots of the terminal after running the tests?

**Comment by gaurivn** [comment]
> @gaurivn could u pls help me, I understand that the test pull request guide should contain
> 
> 1. Steps to test a PR
> 2. Template to report PR testing results
> 3. Steps to test multiple features (I cannot understand what I need to add in this part )
>    Do u mean that need to run each test using **python manage.py test** & provide the screenshots of the terminal after running the tests?

@ashwinijha6, no need of points 1 and 2 you mentioned since that's being handled in a separate PR. I will update issue title and description in some time. Not testing, debugging to be precise, do run VMS locally and signup/login as admin and try see if you can perform various tasks mentioned earlier.

**Comment by ashwinijha6** [comment]
> > @gaurivn could u pls help me, I understand that the test pull request guide should contain
> > 
> > 1. Steps to test a PR
> > 2. Template to report PR testing results
> > 3. Steps to test multiple features (I cannot understand what I need to add in this part )
> >    Do u mean that need to run each test using **python manage.py test** & provide the screenshots of the terminal after running the tests?
> 
> @ashwinijha6, no need of points 1 and 2 you mentioned since that's being handled in a separate PR. I will update issue title and description in some time. Not testing, debugging to be precise, do run VMS locally and signup/login as admin and try see if you can perform various tasks mentioned earlier.

Ok sure @gaurivn

**Comment by ashwinijha6** [comment]
> > @gaurivn could u pls help me, I understand that the test pull request guide should contain
> > 
> > 1. Steps to test a PR
> > 2. Template to report PR testing results
> > 3. Steps to test multiple features (I cannot understand what I need to add in this part )
> >    Do u mean that need to run each test using **python manage.py test** & provide the screenshots of the terminal after running the tests?
> 
> @ashwinijha6, no need of points 1 and 2 you mentioned since that's being handled in a separate PR. I will update issue title and description in some time. Not testing, debugging to be precise, do run VMS locally and signup/login as admin and try see if you can perform various tasks mentioned earlier.

@gaurivn u make the changes required in the description of the issue I will try & make a pull request by today

**Comment by vaibhavarora102** [comment]
hi @gaurivn 
i would like to do this as  only this type  of category i missed during OSH
and would try my best to complete it as soon as possible
 kindly assign me as soon as possible as only one day is left
thanks
